src/chainparams.cpp

```diff
@@ -108,6 +108,7 @@ class CElementsParams : public CChainParams {
         consensus.nPowTargetSpacing = 10 * 60;
         consensus.fPowAllowMinDifficultyBlocks = false;
         consensus.fPowNoRetargeting = false;
+        consensus.nMinDepth = GetArg("-peginconfirmationdepth", DEFAULT_PEGIN_CONFIRMATION_DEPTH);
```

Comment:
consensus.nMinDepth could be committed in the genesis block if it's a consensus rule. Also, can you add this to CCustomParams::UpdateFromArgs and use the same naming scheme?
Perhaps CElementsParams and CRegTestParams could both inherit from CCustomParams? Let's leave that for another PR, I guess.

Reply:
there are reasons you might want to change this(increase in the case of suspicion of future reorgs). Good news is after a certain amount of time, all values here end up under the same consensus.

Reply:
It's more of a "soft" consensus rule than a hard one.

If it can be set to different values over different times in a chain's history, and clients in the future will never notice the older values, I don't see the value of committing to it.

Reply:
Why not? This was chains starting with different values will have different genesis blocks.
Adding it to CCustomParams::UpdateFromArgs is orthogonal, can we do that here?